test/integration/networks/networks_test.go

```diff
@@ -350,15 +350,15 @@ func TestNetworks(t *testing.T) {
 					// networks.DefaultVerify(assert)
 
 					servicePerimeterLink := fmt.Sprintf("accessPolicies/%s/servicePerimeters/%s", policyID, networks.GetStringOutput("restricted_service_perimeter_name"))
-					accessLevel := fmt.Sprintf("accessPolicies/%s/accessLevels/%s", policyID, networks.GetStringOutput("restricted_access_level_name"))
+					accessLevel := fmt.Sprintf("accessPolicies/%s/accessLevels/%s", policyID, networks.GetStringOutput("access_level_name"))
 					networkNames := getNetworkResourceNames(envCode, networkMode, firewallMode)
 
-					servicePerimeter := gcloud.Runf(t, "access-context-manager perimeters describe %s --policy %s", servicePerimeterLink, policyID)
+					servicePerimeter, err := gcloud.RunCmdE(t, fmt.Sprintf("access-context-manager perimeters dry-run describe %s --policy %s", servicePerimeterLink, policyID))
 					assert.Equal(servicePerimeterLink, servicePerimeter.Get("name").String(), fmt.Sprintf("service perimeter %s should exist", servicePerimeterLink))
 					listLevels := utils.GetResultStrSlice(servicePerimeter.Get("status.accessLevels").Array())
 					assert.Contains(listLevels, accessLevel, fmt.Sprintf("service perimeter %s should have access level %s", servicePerimeterLink, accessLevel))
 					listServices := utils.GetResultStrSlice(servicePerimeter.Get("status.restrictedServices").Array())
-					assert.Subset(listServices, restrictedServices, fmt.Sprintf("service perimeter %s should restrict all supported services", servicePerimeterLink))
+					assert.Containsf(t, listServices, restrictedServices, fmt.Sprintf("service perimeter %s should restrict all supported services", servicePerimeterLink))
```

Comment:
suggestion for the test:
```hcl
			servicePerimeter, err := gcloud.RunCmdE(t, fmt.Sprintf("access-context-manager perimeters dry-run describe %s --policy %s", servicePerimeterLink, policyID))
			assert.NoError(err)
			assert.True(strings.Contains(servicePerimeter, servicePerimeterLink), fmt.Sprintf("service perimeter %s should exist", servicePerimeterLink))
			assert.True(strings.Contains(servicePerimeter, accessLevel), fmt.Sprintf("service perimeter %s should have access level %s", servicePerimeterLink, accessLevel))
			for _, service := range restrictedServices {
				assert.True(strings.Contains(servicePerimeter, service), fmt.Sprintf("service perimeter %s should restrict all supported services", servicePerimeterLink))
			}
```

`servicePerimeter.Get("status.accessLevels").Array()` and `servicePerimeter.Get("name").String()`
do not work because they depend on the result being json 
we need to use `strings.Contains(servicePerimeter, accessLevel)` to check the values

Also, there is no Ingres and Egress rules on the output of the gcloud command, the section that is testing those should be commented or removed
https://github.com/eeaton/terraform-example-foundation/blob/fix-1209-vpcsc-dryrun/test/integration/networks/networks_test.go#L363C6-L371C1

Reply:
Got it thanks, I was trying assert.containsF() to try to use the boolean result of a contains method, but I lke your suggestion of wrapping the strings.contains() in assert.true(). 

Also, when I tried an earlier version with `servicePerimeter, err` I was getting an error like "err is not defined". Might have been conflated with other issues though. 

I'll try this.